datafusion/physical-expr/src/sort_expr.rs

```diff
@@ -107,19 +147,94 @@ impl std::fmt::Display for PhysicalSortRequirement {
 }
 
 impl PhysicalSortRequirement {
+    /// Creates a new `exact` requirement, which must match the
+    /// required options and expression. See
+    /// [`PhysicalSortRequirement`] for examples.
+    pub fn new_exact(expr: Arc<dyn PhysicalExpr>, options: SortOptions) -> Self {
+        Self {
+            expr,
+            options: Some(options),
+        }
+    }
+
+    /// Creates a new `expr_only` requirement, which must match the
+    /// required expression. See [`PhysicalSortRequirement`] for
+    /// examples.
+    pub fn new_expr_only(expr: Arc<dyn PhysicalExpr>) -> Self {
+        Self {
+            expr,
+            options: None,
+        }
+    }
```

Comment:
You could have a single `new` and the caller can pass the second argument as `None` if they don't want to constrain sort options. I get the sense that you have two `new`s because you think it is more readable?

Reply:
The alternative I had in mind was
```rust
for partition_by in partition_by_exprs {
    sort_reqs.push(PhysicalSortRequirement::new(partition_by.clone(), None))
    // sort_reqs.push(PhysicalSortRequirement::new(partition_by.clone(), Some(sort_options)))
}
```

Reply:
I can make that change if you prefer it

Reply:
Changed in [8a073c4](https://github.com/apache/arrow-datafusion/pull/5863/commits/8a073c4ae2252f30c91120cb5e9e3ff3fe540f91)